src/app/leaves/components/LeaveRequestDetails.tsx

```diff
@@ -0,0 +1,77 @@
+import React from "react";
+import {
+  Dialog,
+  DialogContent,
+  DialogHeader,
+  DialogTitle,
+} from "@/components/ui/dialog";
+import { LeavesStatusBadge } from "@/common/StatusBadge/LeavesStatusBadge";
+import { LeaveRequestsTypes, LeavesRequestStatus } from "@/types";
+import { Separator } from "@/components/ui/separator";
+import {
+  FileText as FileTextIcon,
+  Calendar as CalendarIcon,
+  User as UserIcon,
+} from "lucide-react";
+
+interface Props {
+  isOpenViewModal: boolean;
+  setIsOpenViewModal: (state: boolean) => void;
+  leaveRequestDetails: LeaveRequestsTypes;
+}
+
+export const LeaveRequestDetails = ({
+  isOpenViewModal,
+  setIsOpenViewModal,
+  leaveRequestDetails,
+}: Props) => {
+  return (
+    <Dialog open={isOpenViewModal} onOpenChange={setIsOpenViewModal}>
+      <DialogContent>
+        <DialogHeader>
+          <DialogTitle className="text-table-thead text-sm">
+            Leave Request Details
+          </DialogTitle>
+          <div className="flex justify-between items-center py-4">
+            <LeavesStatusBadge
+              status={leaveRequestDetails.status as LeavesRequestStatus}
```

Comment:
@nadirhussain786  why using ass yiu already not defined type

Reply:
Fixed